### PR TITLE
Fix UTs caused by .NET 8 release by using specific nuget version

### PR DESCRIPTION
**Pull request by sebastien-marichal**

### CODE REVIEW

**Review by antonioaversa — APPROVED** [review]
LGTM

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/UriShouldNotBeHardcodedTest.cs [hunk 1]
```diff
@@ -64,7 +64,7 @@ public class UriShouldNotBeHardcodedTest
 
         [DataTestMethod]
         [DataRow("2.0.4", "2.0.3", "2.1.1")]
-        [DataRow(Constants.NuGetLatestVersion, Constants.NuGetLatestVersion, Constants.NuGetLatestVersion)]
```
Comment (andrei-epure-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/UriShouldNotBeHardcodedTest.cs:67]: will you open a ticket to bring this back to NuGet Latest?

why: we keep LatestVersion to be aware if some API changes lead to our rules creating FPs or FNs  (e.g. a deprecated attribute might be removed, so we won't raise anymore on net8 and we need to update the rule)

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/UnusedPrivateMemberTest.cs [hunk 2]
```diff
@@ -150,7 +150,7 @@ public class SkipMigration : Migration
 
         protected override void Up(MigrationBuilder migrationBuilder) { }
     }
-}").AddReferences(EntityFrameworkCoreReferences(Constants.NuGetLatestVersion)).Verify();
```
Comment (andrei-epure-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/UnusedPrivateMemberTest.cs:153]: will you open a ticket to bring this back to NuGet Latest?

For example, will we still ignore migrations on Net 8?

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/PermissiveCorsTest.cs [hunk 3]
```diff
@@ -76,7 +76,7 @@ public class PermissiveCorsTest
                 .Verify();
 
         private static IEnumerable<MetadataReference> AdditionalReferences =>
-            NuGetMetadataReference.MicrosoftNetHttpHeaders(Constants.NuGetLatestVersion)
+            NuGetMetadataReference.MicrosoftNetHttpHeaders("2.1.14")
```
Comment (andrei-epure-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/Hotspots/PermissiveCorsTest.cs:79]: will you open a ticket to bring this back to NuGet Latest?

why: we keep LatestVersion to be aware if some API changes lead to our rules creating FPs or FNs  (e.g. a deprecated attribute might be removed, so we won't raise anymore on net8 and we need to update the rule)

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/UriShouldNotBeHardcodedTest.cs (hunk 1 above)
Comment (sebastien-marichal) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/UriShouldNotBeHardcodedTest.cs:67]: I created the ticket #8363

The goal would be to revert this PR when changing the target framework of the UT project to `net8.0`

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/UnusedPrivateMemberTest.cs (hunk 2 above)
Comment (sebastien-marichal) [reply analyzers/tests/SonarAnalyzer.UnitTest/Rules/UnusedPrivateMemberTest.cs:153]: No we will revert this PR during the migration of UT project to `net8.0`

**Pull request merged**